### PR TITLE
Provide standard build environment using docker

### PR DESCRIPTION
**Pull request by fredericmorin**
To compile klipper for a specific target:
```
./scripts/docker-with-builder-env.sh make menuconfig
./scripts/docker-with-builder-env.sh make -j6
```

To run CI tests:
```
./scripts/docker-with-builder-env.sh ./scripts/ci-build.sh
```

### CODE REVIEW

File: scripts/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+        flex \
+        gcc-arm-none-eabi \
+        gcc-avr \
+        libffi-dev \
+        libgmp-dev \
+        libmpc-dev \
+        libmpfr-dev \
+        libnewlib-arm-none-eabi \
+        pv \
+        python2-dev \
```
Comment (thijstriemstra) [inline scripts/Dockerfile:33]: python2?

File: scripts/Dockerfile [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+# Install gnupru toolchain
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        curl \
+        git \
+        wget \
+    && rm -rf /var/lib/apt/lists/*
+RUN cd /usr/local/src/ && \
+    git config --global user.email "you@example.com" && \
+    git config --global user.name "Your Name" && \
+    git clone https://github.com/dinuxbg/gnupru -b 2023.01 --depth 1 && \
```
Comment (thijstriemstra) [inline scripts/Dockerfile:46]: does this need to be updated with every releaes?

**Review by fredericmorin — COMMENTED** [review]
If I may add a bit of colour to my original PR description, this is a pattern we use ourselves at work, using docker to standardize the build environment of a project across dev machines and ci environment.

File: scripts/Dockerfile (hunk 1 above)
Comment (thijstriemstra) [reply scripts/Dockerfile:33]: > I would much prefer see this move to python3 instead

What is needed for that?

Comment (fredericmorin) [reply scripts/Dockerfile:33]: Happy to look into that. Will give that a try.

**Comment by KevinOConnor** [comment]
Thanks.  As high-level feedback, I'm not really sure what the intended audience of this change is.  That is, who will use it and what tasks will they accomplish with it.  As such, I'm not sure I'm able to review this PR.

Separately, the changes to .gitignore don't look correct.

-Kevin

**Pull request closed without merging**